cpp/dolfinx/generation/RectangleMesh.h

```diff
@@ -12,6 +12,21 @@
 #include <mpi.h>
 #include <string>
 
+namespace dolfinx::generation
+{
+
+/// Enum for different diagonal types
+enum class DiagonalType : int
```

Comment:
Is `int` required here?

Reply:
I could not get the code to compile without int